src/ember-app.js

```diff
@@ -236,10 +243,11 @@ class EmberApp {
   async _visit(path, fastbootInfo, bootOptions, result, buildSandboxPerVisit) {
     let shouldBuildApp = buildSandboxPerVisit || this._applicationInstance === undefined;
 
-    let { app, isSandboxPreBuilt } = shouldBuildApp
+    let { app, appAnalytics } = shouldBuildApp
```

Comment:
Can we call this `analytics` instead of `appAnalytics`?

Reply:
Also, can we unify these? Can we funnel both cases through a single function (e.g. `buildAppInstance` or even `getNewApplicationInstance`) so the rest of the code doesn't need to care about the two pathways?

Reply:
do you mean 
``` let { app, appAnalytics } = shouldBuildApp
      ? await this.getNewApplicationInstance()
      : this.getAppInstanceInfo(this._applicationInstance);
```
should instead be
```
let { app, appAnalytics } = await this.getNewApplicationInstance(shouldBuildApp)
```
and the function would return the appropriate response

Reply:
Yes, IMHO we should have only a single function that is responsible for understanding:

* if we should build a new instance (based on `shouldBuildApp`)
* if we should build a new instance when the queue is empty and a new application instance is requested
* how to build the proper set of analytics output

Reply:
agreed! 👍